src/main/java/com/github/yp44/SetPropertyMojo.java

```diff
@@ -73,53 +72,58 @@ public void execute() throws MojoExecutionException {
     }
 
     private String getGitModifiedFiles() throws MojoExecutionException {
-        final File gitPath = project.getBasedir().toPath().resolve(".git").toFile();
+        final MavenProject root = this.findRoot();
 
+        final String moduleName = project.getBasedir().getName();
+        final File gitPath = root.getBasedir().toPath().resolve(".git").toFile();
+        getLog().info("Git path " + gitPath.getAbsolutePath() + "   base dir " + project.getBasedir().getName());
         final FileRepositoryBuilder builder = new FileRepositoryBuilder();
-        final Repository repo;
-        try {
-            repo = builder.setGitDir(gitPath).build();
-            Git git = new Git(repo);
+
+        try (final Repository repo = builder.setGitDir(gitPath).build();
+             Git git = new Git(repo)) {
 
             final StatusCommand status = git.status();
             final Status call = status.call();
 
-            final StringBuilder files = new StringBuilder();
-
-            Stream<String> stream = Stream.empty();
-            List<String> elementsSource = Arrays.asList(gitStatusElements.split(","));
-
-            for (String method : elementsSource) {
-                try {
-                    method = method.substring(0, 1).toUpperCase() + method.substring(1);
-                    final Method m = call.getClass().getMethod("get" + method);
-                    Set<String> ff = (Set<String>) m.invoke(call);
-                    stream = concat(stream, ff.stream());
-                    getLog().debug("Get git status " + method + " list : " + ff.stream().map(this::toAbsolute).map(f -> "\n" + f).reduce("", (a, f) -> a + f));
-                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
-                    e.printStackTrace();
-                }
-            }
-
-            stream.filter(this::accept)
-                    .map(this::toAbsolute)
-                    .forEach(f -> files.append(f).append(","));
-            String sfiles = files.toString();
+            String sfiles = Arrays.stream(gitStatusElements.split(","))
+                    .map((String name) -> retrieveFiles(call, name))
+                    .flatMap(Set::stream)
+                    .filter(this::accept)
+                    .distinct()
+                    .map((String rel) -> this.toAbsolute(rel, root))
+                    .collect(Collectors.joining(","));
 
-            if (!sfiles.isEmpty()) {
-                sfiles = sfiles.substring(0, sfiles.length() - 1);
+            if (sfiles.isEmpty()) {
+                sfiles = "unexistingFile0001.xml";
```

Comment:
Should use "emptyListValue" option value

Reply:
done